src/Appender/RollingFileAppender.cs

```diff
@@ -1695,6 +1708,8 @@ protected DateTime NextCheckDate(DateTime currentDateTime, RollPoint rollPoint)
 		/// A mutex that is used to lock rolling of files.
 		/// </summary>
 		private Mutex m_mutexForRolling;
+
+		private string m_mutexNameForRolling = null;
```

Comment:
Storing this as an internal state appears to be unnecessary, isn't it?

Reply:
I would remove it, but consider my following experience:
My reason to introduce the `m_mutexNameForRolling` was tracking the change of log file path  of the `RollingFileAppender`, after it was first correctly activated, via assigning the new file path and calling Activate() on the appender in user code again. I experienced mutex leak while recreating the mutex with a different Id, so I wanted to track this down.

Background: I know a customer doing stuff like changing the log file path in run time, due to logging to a different specific directory, based on some global DB settings, but wanting also to track other potential startup issues before the log file path was changed....

Reply:
Such a functional requirement is better implemented by streaming the logs into a database or to a remote location. That remote location could then buffer log events in a safe manner while it is being reconfigured. The rolling file appender is, as far as I can tell, not made to handle this.

Reply:
ok, good to know. Thanks for clarifying this to me. I do not have much experience with streaming logs into a DB, at least not with log4net. Log4net file or console logging is for me always the "last and securest resort" to trace things in the weird cases, when you don't see any SQL or Mail messages or other App output. Log4net always delivered the essential log information to tackle errors in such situations. So, I noted, RollingFileAppender is not good to change after its first activation.